src/app/layout/header/header.component.ts

```diff
@@ -77,74 +49,199 @@ export class HeaderComponent implements OnInit, OnDestroy {
+    if (context.space) {
+      this.logger.log('[HeaderComponent] Switching current space to ' + context.space.id);
+      // go to internal route of this context
+      this.router.navigate([context.path]);
+    }
+  }
 
+  private onSelectViewAllSpaces(): void {
+    this.goToInternal('/' + this.loggedInUser.attributes.username + '/_spaces');
   }
 
-  onImgLoad() {
-    this.imgLoaded = true;
+  private onSelectAccountHome(): void {
+    this.goToInternal('/_home');
   }
 
-  resetData(): void {
-    this.imgLoaded = false;
+  private onSelectUserProfile(): void {
+    this.goToInternal('/' + this.loggedInUser.attributes.username);
+  }
+
+  private onFollowedLink(url: string): void {
+    // NOP
+  }
+
+  ngOnInit(): void {
+
+    // logout can also be called by an event from other parts of the app
+    this.broadcaster.on<string>('logout')
+      .subscribe(message => {
+        this.logger.warn('[HeaderComponent] Received logout broadcast event.');
+        this.onSelectLogout();
+      });
+
+    // on an authentication error, we logout and send the user to the login
+    this.broadcaster.on<any>('authenticationError')
+      .subscribe(() => {
+        this.logger.warn('[HeaderComponent] Received authenticationError broadcast event.');
+        this.onSelectLogout();
+        this.router.navigate(['/login']);
+      });
+
+    // we subscribe to the UserService to get notified when the user switches
+    this.userService.getUser().subscribe(user => {
+      if (user && user.id) {
+        this.logger.log('[HeaderComponent] UserService signals new user ' + user.attributes.username);
+        this.loggedInUser = user;
+      } else {
+        this.logger.warn('[HeaderComponent] UserService returned empty object user value.');
+        this.loggedInUser = null;
+      }
+    });
+
+    // we subscribe to the current space to get notified when the space switches. This only fires if a switch is happening, not on bootstrap
+    this.contexts.current.subscribe(val => {
+      this.logger.log('[HeaderComponent] Received new currentContext: ' + val ? val.name : 'null');
+      this.currentContext = val;
+    });
+
+    // we subscribe to all spaces list to get notified when the spaces list changes
+    this.contexts.recent.subscribe(val => {
+      this.logger.log('[HeaderComponent] Received from contexts new recent contexts list with length: ' + val.length);
+      // TODO: context.recent seems to be only sending one Context at a time. Is this correct?
+      // TODO: the following is a hack, replace it!
+      if (val) {
+        val.forEach(element => {
+          if (!this.alreadyInRecent(element.path)) {
+            this.headerService.addRecentContext(element);
+            this.recentContexts.push(element);
+          }
+        });
+      }
+      // if there is no currentContext yet, we select the first one to be the new currentContext
+      if (!this.currentContext) {
+        this.currentContext = this.recentContexts[0];
```

Comment:
Can we ensure the there even is one recent context all the time?

Reply:
No, and this is one of the open integration issues. That setup here is wrong and should be fixed.